### PR TITLE
Bp 2060 prestashop plugin improvments update

### PR DESCRIPTION
**Pull request by vegimcarkaxhija**

### CODE REVIEW

File: api/paymentmethods/billink/billink.php [hunk 1]
```diff
@@ -119,14 +119,16 @@ public function getPayload($products)
...
         if (self::CUSTOMER_TYPE_B2C != $this->CustomerType) {
-            $payload['billing']['recipient']['companyName'] = $this->BillingCompanyName;;
-            $payload['billing']['recipient']['chamberOfCommerce'] = $this->CompanyCOCRegistration;
-
-            if (isset($payload['shipping'])){
-                $payload['shipping']['recipient']['companyName'] = $this->ShippingCompanyName;
-                $payload['shipping']['recipient']['category'] = 'B2B';
+            if($this->BillingCompanyName){
+                $payload['billing']['recipient']['careOf'] = $this->BillingCompanyName;
+                $payload['billing']['recipient']['chamberOfCommerce'] = $this->CompanyCOCRegistration;
```
Comment (harli91) [inline api/paymentmethods/billink/billink.php:126]: this seems somewhat ugly to me why not use array_merge_recursive?

File: buckaroo3.php [hunk 2]
```diff
@@ -1546,10 +1593,6 @@ public function hookDisplayAdminProductsMainStepLeftColumnMiddle($params)
      */
     protected function getAddressById($id)
     {
-        if (is_int($id)) {
-            return;
-        }
-
```
Comment (harli91) [inline buckaroo3.php:1552]: why did you remove this check?

File: controllers/front/ajax.php [hunk 3]
```diff
@@ -46,8 +46,20 @@ public function postProcess()
...
+                if (strpos($paymentFee, '%') !== false) {
+                    $paymentFee = str_replace('%', '', $paymentFee);
+                    $paymentFee = new Number((string)$paymentFee);
+                    $percentage = $paymentFee->dividedBy(new Number('100'));
+                    $paymentFee = $orderTotal->times($percentage);
+                } else if ($paymentFee > 0) {
+                    // The fee is a flat amount.
+                    $paymentFee = new Number((string)$paymentFee);
+                }
+                $buckarooFee = Tools::displayPrice($paymentFee->toPrecision(2));
```
Comment (harli91) [inline controllers/front/ajax.php:61]: these functionality is used in another class, maybe extract this in a separate service class

File: views/js/buckaroo.admin.js [hunk 4]
```diff
@@ -36,4 +36,44 @@ $(document).ready(function () {
...
+        var panelID = this.dataset.target;
+        // Check if the switch is enabled (checked)
+        $(panelID).collapse('toggle');
+    });
+    $( function() {
+        $( "#sortable" ).sortable();
+        $( "#sortable" ).disableSelection();
+    } );
+
+    function updatePositions() {
```
Comment (harli91) [inline views/js/buckaroo.admin.js:62]: better give it a distinct name, may conflict with other similar named function

**Review by harli91 — APPROVED** [review]

File: views/templates/front/error.tpl [hunk 5]
```diff
@@ -21,7 +21,7 @@
         <a href="index.php?controller=order&submitReorder=&id_order={$order_id|escape:'html':'UTF-8'}"
            class="button btn btn-default standart-checkout button-medium"><span>{l s='Return to checkout' mod='buckaroo3'}</span></a>
     {else}
-        <a href="index.php?controller=order"
+        <a href="order"
```
[the remark this answers is not in the document]
Comment (vegimcarkaxhija) [reply views/templates/front/error.tpl:24]: if you enable debug, you get a warning to not use that absolute url and switch the url to order, it is update in Prestashop 8

File: buckaroo3.php [hunk 6]
```diff
@@ -279,8 +280,9 @@ private function installTab()
         $parent_tab = new Tab();
         $parent_tab->name[$this->context->language->id] = $this->l('Buckaroo Payments');
         $parent_tab->class_name = 'AdminBuckaroo';
-        $parent_tab->id_parent = 0;
+        $parent_tab->id_parent = Tab::getIdFromClassName('IMPROVE');
```
[the remark this answers is not in the document]
Comment (vegimcarkaxhija) [reply buckaroo3.php:283]: it will get the id from IMPROVE menu, so our module will be placed in that menu

File: views/js/buckaroo.admin.js [hunk 7]
```diff
@@ -36,4 +36,44 @@ $(document).ready(function () {
...
+    $(".toggle-switch").change(function() {
+        // Get the ID of the panel associated with this switch
+        var panelID = this.dataset.target;
+        // Check if the switch is enabled (checked)
+        $(panelID).collapse('toggle');
+    });
+    $( function() {
+        $( "#sortable" ).sortable();
+        $( "#sortable" ).disableSelection();
+    } );
```
[the remark this answers is not in the document]
Comment (vegimcarkaxhija) [reply views/js/buckaroo.admin.js:60]: $(function() { ... });
is just jQuery short-hand for
$(document).ready(function() { ... });
"This is the jQuery ready function. It ensures that your code doesn't run until the document is fully loaded, which is necessary when manipulating DOM elements."

File: views/css/buckaroo3.admin.css [hunk 8]
```diff
@@ -10,4 +10,70 @@ legend.active {
...
+}
+
+.slider.round:before {
+    border-radius: 50%;
+}
+.button-group {
+    display: flex;
+    justify-content: space-between;  /* optional, if you want to space your buttons evenly */
+}
+
```
[the remark this answers is not in the document]
Comment (vegimcarkaxhija) [reply views/css/buckaroo3.admin.css]: changed from slider to buckaroo-slider and from switch to buckaroo-switch

File: views/templates/front/error.tpl (hunk 5 above)
[the remark this answers is not in the document]
Comment (harli91) [reply views/templates/front/error.tpl:24]: what kind of warning? depreciation?

File: views/js/buckaroo.admin.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (harli91) [reply views/js/buckaroo.admin.js:60]: don't you have a document ready function wrapping the whole code already ? https://github.com/buckaroo-it/PrestaShop/blob/cdcc65bb9312b8d745eef1f75d2d4b814e45807d/views/js/buckaroo.admin.js#L15

File: views/css/buckaroo3.admin.css (hunk 8 above)
[the remark this answers is not in the document]
Comment (harli91) [reply views/css/buckaroo3.admin.css]: what about the css for the legend ?

File: views/js/buckaroo.admin.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (vegimcarkaxhija) [reply views/js/buckaroo.admin.js:60]: yeah you're right
